src/app/hooks/use-search-parameters-validation.tsx

```diff
@@ -0,0 +1,381 @@
+
+		await coin.__construct();
+
+		const isValid = await coin.address().validate(recipient);
+
+		if (!isValid) {
+			return { error: { type: SearchParametersError.NetworkMismatch } };
+		}
+	}
+};
+
+/* istanbul ignore next */
+const WrapperQR = ({ children }) => <span>Invalid QR: {children}</span>;
+/* istanbul ignore next */
+const WrapperURI = ({ children }) => <span>Invalid URI: {children}</span>;
+
+export const useSearchParametersValidation = () => {
+	const methods = {
+		transfer: {
+			path: ({ profile, searchParameters }: PathProperties) =>
+				`${generatePath(ProfilePaths.SendTransfer, {
+					profileId: profile.id(),
+				})}?${searchParameters.toString()}`,
+			validate: validateTransfer,
+		},
+		vote: {
+			path: ({ profile, searchParameters, env }: PathProperties) => {
+				const network = findNetworkFromSearchParameters(profile, searchParameters);
+				assertNetwork(network);
+
+				const delegate = delegateFromSearchParameters({ env, network, profile, searchParameters });
+
+				searchParameters.set("vote", delegate?.address() as string);
+
+				return `${generatePath(ProfilePaths.SendVote, {
+					profileId: profile.id(),
+				})}?${searchParameters.toString()}`;
+			},
+			validate: validateVote,
+		},
+	};
+
+	const validateSearchParameters = async (
+		profile: Contracts.IProfile,
+		env: Environment,
+		parameters: URLSearchParams,
+		requiredParameters?: RequiredParameters,
+	) => {
+		assertProfile(profile);
+
+		const allEnabledNetworks = profileAllEnabledNetworks(profile);
+
+		const coin = parameters.get("coin")?.toUpperCase() || "ARK";
+		const method = parameters.get("method")?.toLowerCase();
+		const networkId = parameters.get("network")?.toLowerCase();
+		const nethash = parameters.get("nethash");
+
+		if (!networkId && !nethash) {
+			return { error: { type: SearchParametersError.MissingNetworkOrNethash } };
+		}
+
+		if (!method) {
+			return { error: { type: SearchParametersError.MissingMethod } };
+		}
+
+		if (requiredParameters?.coin && coin !== requiredParameters?.coin) {
+			return { error: { type: SearchParametersError.CoinMismatch } };
+		}
+
+		if (!allEnabledNetworks.some((item) => item.coin() === coin)) {
+			return { error: { type: SearchParametersError.CoinNotSupported, value: coin } };
+		}
+
+		if (!Object.keys(methods).includes(method)) {
+			return { error: { type: SearchParametersError.MethodNotSupported, value: method } };
+		}
+
+		let network: Networks.Network | undefined;
+
+		if (networkId) {
+			if (requiredParameters?.network && networkId !== requiredParameters?.network) {
+				return { error: { type: SearchParametersError.NetworkMismatch } };
+			}
+
+			if (!defaultNetworks[networkId]) {
+				return { error: { type: SearchParametersError.NetworkInvalid, value: networkId } };
+			}
+
+			network = profile.availableNetworks().find((item) => item.id() === networkId);
+
+			if (network && !network.meta().enabled) {
```

Comment:
@dated I think we could exclude default networks from this condition because their `meta().enabled` will most probably return false. something like:
```suggestion
			if (network && !network.meta().enabled && !defaultNetworks[networkId]) {

```